src/object/custom_particle_system.cpp

```diff
@@ -1241,4 +1246,1024 @@ CustomParticleSystem::ease_value(float* value, float target, float time, easing
   script_easings.push_back(req);
 }
 
+void
+CustomParticleSystem::spawn_particles(int amount, bool instantly)
+{
+  if (instantly)
+  {
+    for (int i  = 0; i < amount; i++)
+    {
+      spawn_particles(0.f);
+    }
+  }
+  else
+  {
+    // TODO: Implement delayed spawn mode for scripting.
+    log_warning << "Delayed spawn mode is not yet implemented for scripting." << std::endl;
+  }
+}
+
+// =============================================================================
+// ============================   ATTRIBUTES   =================================
+// =============================================================================
+
+int
+CustomParticleSystem::get_max_amount() const
+{
+  return m_max_amount;
+}
+
+void
+CustomParticleSystem::set_max_amount(int amount)
+{
+  m_max_amount = amount;
+}
+
+bool
+CustomParticleSystem::get_cover_screen() const
+{
+  return m_cover_screen;
+}
+
+void
+CustomParticleSystem::set_cover_screen(bool cover)
+{
+  m_cover_screen = cover;
+}
+
+std::string
+CustomParticleSystem::get_birth_mode() const
+{
+  switch (m_particle_birth_mode)
+  {
+  case CustomParticleSystem::FadeMode::None:
+    return "None";
+
+  case CustomParticleSystem::FadeMode::Fade:
+    return "Fade";
+
+  case CustomParticleSystem::FadeMode::Shrink:
+    return "Shrink";
+  
+  default:
+    return "";
+  }
+}
+
+void
+CustomParticleSystem::set_birth_mode(const std::string& mode)
+{
+  if (mode == "None")
+  {
+    m_particle_birth_mode = CustomParticleSystem::FadeMode::None;
+  }
+  else if (mode == "Fade")
+  {
+    m_particle_birth_mode = CustomParticleSystem::FadeMode::Fade;
+  }
+  else if (mode == "Shrink")
+  {
+    m_particle_birth_mode = CustomParticleSystem::FadeMode::Shrink;
+  }
+  else
+  {
+    log_warning << "Invalid option " + mode + "; valid options are: None, Fade, Shrink." << std::endl;
+  }
+}
+
+std::string
+CustomParticleSystem::get_death_mode() const
+{
+  switch (m_particle_death_mode)
+  {
+  case CustomParticleSystem::FadeMode::None:
+    return "None";
+
+  case CustomParticleSystem::FadeMode::Fade:
+    return "Fade";
+
+  case CustomParticleSystem::FadeMode::Shrink:
+    return "Shrink";
+  
+  default:
+    return "";
+  }
+}
+
+void
+CustomParticleSystem::set_death_mode(const std::string& mode)
+{
+  if (mode == "None")
+  {
+    m_particle_death_mode = CustomParticleSystem::FadeMode::None;
+  }
+  else if (mode == "Fade")
+  {
+    m_particle_death_mode = CustomParticleSystem::FadeMode::Fade;
+  }
+  else if (mode == "Shrink")
+  {
+    m_particle_death_mode = CustomParticleSystem::FadeMode::Shrink;
+  }
+  else
+  {
+    log_warning << "Invalid option " + mode + "; valid options are: None, Fade, Shrink." << std::endl;
+  }
+}
+
+std::string
+CustomParticleSystem::get_rotation_mode() const
+{
+  switch (m_particle_rotation_mode)
+  {
+  case CustomParticleSystem::RotationMode::Fixed:
```

Comment:
Same for RotationMode, introduce that in a using statement at the start.

Reply:
In these cases, the `CustomParticleSystem::` part could simply be omitted, since these are functions of the class.